<a href="https://colab.research.google.com/github/ejr271292/Proyecto/blob/main/ProyectoDS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [29]:
#Importamos Pandas Módulo:
import pandas as pd

In [30]:
#Cargamos el dataset:
from google.colab import drive
import os
drive.mount('/content/gdrive')


Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [31]:
df=pd.read_excel('/content/gdrive/MyDrive/Proyecto Final Data Science 1/Estadisticas biocombustibles.xls')

In [32]:
#Información del df:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5741 entries, 0 to 5740
Data columns (total 25 columns):
 #   Column                                 Non-Null Count  Dtype         
---  ------                                 --------------  -----         
 0   PERÍODO                                5741 non-null   datetime64[ns]
 1   EMPRESA                                5741 non-null   object        
 2   LOCALIDAD                              5741 non-null   object        
 3   PROVINCIA                              5741 non-null   object        
 4   CUPO TOTAL ASIGNADO                    5417 non-null   float64       
 5   PRODUCCIÓN                             5737 non-null   float64       
 6   VENTAS AL CORTE                        5741 non-null   float64       
 7   OTRAS VENTAS                           5741 non-null   float64       
 8   AXION ENERGY ARGENTINA S.A.            5741 non-null   float64       
 9   DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  5741 non-null   float64 

In [33]:
#Tamaño del df:
df.shape



(5741, 25)

In [34]:
#Verifico que tipos de datos tengo para cada variable:
df.dtypes
print('Los datos son de tipo:\n',df.dtypes)

Los datos son de tipo:
 PERÍODO                                  datetime64[ns]
EMPRESA                                          object
LOCALIDAD                                        object
PROVINCIA                                        object
CUPO TOTAL ASIGNADO                             float64
PRODUCCIÓN                                      float64
VENTAS AL CORTE                                 float64
OTRAS VENTAS                                    float64
AXION ENERGY ARGENTINA S.A.                     float64
DESTILERÍA ARGENTINA DE PETRÓLEO S.A.           float64
PETROBRAS ARGENTINA S.A.                        float64
PETROLERA DEL CONOSUR S.A.                      float64
POLIPETROL S.A.                                 float64
REFINERÍA DEL NORTE S.A.                        float64
SHELL ARGENTINA C.A.P.S.A.                      float64
YPF S.A.                                        float64
OIL COMBUSTIBLES S.A.                           float64
NEW AMERICAN OIL S.A.   

In [35]:
#Columnas del df:
df.columns

Index(['PERÍODO', 'EMPRESA', 'LOCALIDAD', 'PROVINCIA', 'CUPO TOTAL ASIGNADO ',
       'PRODUCCIÓN ', 'VENTAS AL CORTE ', 'OTRAS VENTAS ',
       'AXION ENERGY ARGENTINA S.A.', 'DESTILERÍA ARGENTINA DE PETRÓLEO S.A.',
       'PETROBRAS ARGENTINA S.A.', 'PETROLERA DEL CONOSUR S.A.',
       'POLIPETROL S.A.', 'REFINERÍA DEL NORTE S.A.',
       'SHELL ARGENTINA C.A.P.S.A.', 'YPF S.A.', 'OIL COMBUSTIBLES S.A.',
       'NEW AMERICAN OIL S.A.', 'REFI PAMPA S.A.', 'TRAFIGURA ARGENTINA S.A.',
       'ENERGÍA ARGENTINA S.A. (ENARSA)',
       'ENERGÍA Y DERIVADOS DEL PETRÓLEO S.A.', 'PETROIL S.A.',
       'DIVERSE FUELS S.A.', 'PETROLERA DEGAB S.A.'],
      dtype='object')

In [36]:
#Primeras 5 filas/registros:
df.head()


PERÍODO                          EMPRESA        LOCALIDAD     PROVINCIA  \
0 2010-01-01  ADVANCED ORGANIC MATERIALS S.A.            PILAR  BUENOS AIRES   
1 2010-01-01              T 6 INDUSTRIAL S.A.      SAN LORENZO      SANTA FE   
2 2010-01-01                  UNITEC BIO S.A.      SAN LORENZO      SANTA FE   
3 2010-01-01                       PITEY S.A.   VILLA MERCEDES      SAN LUIS   
4 2010-01-01                VICENTÍN S.A.I.C.       AVELLANEDA      SANTA FE   

   CUPO TOTAL ASIGNADO   PRODUCCIÓN   VENTAS AL CORTE   OTRAS VENTAS   \
0                   0.0      687.990               0.0           0.00   
1                   0.0    19175.270               0.0           0.00   
2                   0.0    15696.790               0.0           0.00   
3                   0.0      201.017               0.0         200.86   
4                   0.0     8993.550               0.0           3.53   

   AXION ENERGY ARGENTINA S.A.  DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  ...  \
0                          0.0                                    0.0  ...   
1                          0.0                                    0.0  ...   
2                          0.0                                    0.0  ...   
3                          0.0                                    0.0  ...   
4                          0.0                                    0.0  ...   

   YPF S.A.  OIL COMBUSTIBLES S.A.  NEW AMERICAN OIL S.A.  REFI PAMPA S.A.  \
0       0.0                    0.0                    0.0              0.0   
1       0.0                    0.0                    0.0              0.0   
2       0.0                    0.0                    0.0              0.0   
3       0.0                    0.0                    0.0              0.0   
4       0.0                    0.0                    0.0              0.0   

   TRAFIGURA ARGENTINA S.A.  ENERGÍA ARGENTINA S.A. (ENARSA)  \
0                       0.0                              0.0   
1                       0.0                              0.0   
2                       0.0                              0.0   
3                       0.0                              0.0   
4                       0.0                              0.0   

   ENERGÍA Y DERIVADOS DEL PETRÓLEO S.A.  PETROIL S.A.  DIVERSE FUELS S.A.  \
0                                    0.0           0.0                 0.0   
1                                    0.0           0.0                 0.0   
2                                    0.0           0.0                 0.0   
3                                    0.0           0.0                 0.0   
4                                    0.0           0.0                 0.0   

   PETROLERA DEGAB S.A.  
0                   0.0  
1                   0.0  
2                   0.0  
3                   0.0  
4                   0.0  

[5 rows x 25 columns]

In [37]:
#Modifico el índice del df:
df.set_index("PERÍODO")

EMPRESA        LOCALIDAD  \
PERÍODO                                                                      
2010-01-01                ADVANCED ORGANIC MATERIALS S.A.            PILAR   
2010-01-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2010-01-01                                UNITEC BIO S.A.      SAN LORENZO   
2010-01-01                                     PITEY S.A.   VILLA MERCEDES   
2010-01-01                              VICENTÍN S.A.I.C.       AVELLANEDA   
...                                                   ...              ...   
2024-02-01  SANTA FE GAS Y ENERGÍAS RENOVABLES S.A.P.E.M.         SANTA FE   
2024-02-01                                 SOYENERGY S.A.            PILAR   
2024-02-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2024-02-01                                UNITEC BIO S.A.      SAN LORENZO   
2024-02-01                              VICENTÍN S.A.I.C.       AVELLANEDA   

               PROVINCIA  CUPO TOTAL ASIGNADO   PRODUCCIÓN   VENTAS AL CORTE   \
PERÍODO                                                                         
2010-01-01  BUENOS AIRES              0.000000      687.990              0.00   
2010-01-01      SANTA FE              0.000000    19175.270              0.00   
2010-01-01      SANTA FE              0.000000    15696.790              0.00   
2010-01-01      SAN LUIS              0.000000      201.017              0.00   
2010-01-01      SANTA FE              0.000000     8993.550              0.00   
...                  ...                   ...          ...               ...   
2024-02-01      Santa Fe                   NaN        0.000              0.00   
2024-02-01  Buenos Aires            937.173498      241.510            244.00   
2024-02-01      Santa Fe              0.000000     2865.510              0.00   
2024-02-01      Santa Fe           2602.843196     2925.960           3059.54   
2024-02-01      Santa Fe              0.000000        0.000              0.00   

            OTRAS VENTAS   AXION ENERGY ARGENTINA S.A.  \
PERÍODO                                                  
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01         200.86                          0.0   
2010-01-01           3.53                          0.0   
...                   ...                          ...   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                       1080.5   
2024-02-01           0.00                          0.0   

            DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  PETROBRAS ARGENTINA S.A.  \
PERÍODO                                                                       
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
...                                           ...                       ...   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   

            ...  YPF S.A.  OIL COMBUSTIBLES S.A.  NEW AMERICAN OIL S.A.  \
PERÍODO     ...                                                           
2010-01-01  ...       0.0                    0.0

In [38]:
df

PERÍODO                                        EMPRESA  \
0    2010-01-01                ADVANCED ORGANIC MATERIALS S.A.   
1    2010-01-01                            T 6 INDUSTRIAL S.A.   
2    2010-01-01                                UNITEC BIO S.A.   
3    2010-01-01                                     PITEY S.A.   
4    2010-01-01                              VICENTÍN S.A.I.C.   
...         ...                                            ...   
5736 2024-02-01  SANTA FE GAS Y ENERGÍAS RENOVABLES S.A.P.E.M.   
5737 2024-02-01                                 SOYENERGY S.A.   
5738 2024-02-01                            T 6 INDUSTRIAL S.A.   
5739 2024-02-01                                UNITEC BIO S.A.   
5740 2024-02-01                              VICENTÍN S.A.I.C.   

            LOCALIDAD     PROVINCIA  CUPO TOTAL ASIGNADO   PRODUCCIÓN   \
0               PILAR  BUENOS AIRES              0.000000      687.990   
1         SAN LORENZO      SANTA FE              0.000000    19175.270   
2         SAN LORENZO      SANTA FE              0.000000    15696.790   
3      VILLA MERCEDES      SAN LUIS              0.000000      201.017   
4          AVELLANEDA      SANTA FE              0.000000     8993.550   
...               ...           ...                   ...          ...   
5736         SANTA FE      Santa Fe                   NaN        0.000   
5737            PILAR  Buenos Aires            937.173498      241.510   
5738      SAN LORENZO      Santa Fe              0.000000     2865.510   
5739      SAN LORENZO      Santa Fe           2602.843196     2925.960   
5740       AVELLANEDA      Santa Fe              0.000000        0.000   

      VENTAS AL CORTE   OTRAS VENTAS   AXION ENERGY ARGENTINA S.A.  \
0                 0.00           0.00                          0.0   
1                 0.00           0.00                          0.0   
2                 0.00           0.00                          0.0   
3                 0.00         200.86                          0.0   
4                 0.00           3.53                          0.0   
...                ...            ...                          ...   
5736              0.00           0.00                          0.0   
5737            244.00           0.00                          0.0   
5738              0.00           0.00                          0.0   
5739           3059.54           0.00                       1080.5   
5740              0.00           0.00                          0.0   

      DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  ...  YPF S.A.  \
0                                       0.0  ...       0.0   
1                                       0.0  ...       0.0   
2                                       0.0  ...       0.0   
3                                       0.0  ...       0.0   
4                                       0.0  ...       0.0   
...                                     ...  ...       ...   
5736                                    0.0  ...       0.0   
5737                                    0.0  ...       0.0   
5738                                    0.0  ...       0.0   
5739                                    0.0  ...       0.0   
5740                                    0.0  ...       0.0   

      OIL COMBUSTIBLES S.A.  NEW AMERICAN OIL S.A.  REFI PAMPA S.A.  \
0                       0.0                    0.0              0.0   
1                       0.0                    0.0              0.0   
2                       0.0                    0.0              0.0   
3                       0.0                    0.0              0.0   
4                       0.0                    0.0              0.0   
...                     ...                    ...              ...   
5736                    0.0                    0.0              0.0   
5737                    0.0                    0.0              0.0   
5738                    0.0                    0.0              0.0   
5739                    0.0   

In [39]:
#Como vemos que el df original no fue modificado, si quiero cambiar el índice de forma definitiva debo utilizar inplace:
df.set_index('PERÍODO', inplace=True)

In [40]:
df

EMPRESA        LOCALIDAD  \
PERÍODO                                                                      
2010-01-01                ADVANCED ORGANIC MATERIALS S.A.            PILAR   
2010-01-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2010-01-01                                UNITEC BIO S.A.      SAN LORENZO   
2010-01-01                                     PITEY S.A.   VILLA MERCEDES   
2010-01-01                              VICENTÍN S.A.I.C.       AVELLANEDA   
...                                                   ...              ...   
2024-02-01  SANTA FE GAS Y ENERGÍAS RENOVABLES S.A.P.E.M.         SANTA FE   
2024-02-01                                 SOYENERGY S.A.            PILAR   
2024-02-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2024-02-01                                UNITEC BIO S.A.      SAN LORENZO   
2024-02-01                              VICENTÍN S.A.I.C.       AVELLANEDA   

               PROVINCIA  CUPO TOTAL ASIGNADO   PRODUCCIÓN   VENTAS AL CORTE   \
PERÍODO                                                                         
2010-01-01  BUENOS AIRES              0.000000      687.990              0.00   
2010-01-01      SANTA FE              0.000000    19175.270              0.00   
2010-01-01      SANTA FE              0.000000    15696.790              0.00   
2010-01-01      SAN LUIS              0.000000      201.017              0.00   
2010-01-01      SANTA FE              0.000000     8993.550              0.00   
...                  ...                   ...          ...               ...   
2024-02-01      Santa Fe                   NaN        0.000              0.00   
2024-02-01  Buenos Aires            937.173498      241.510            244.00   
2024-02-01      Santa Fe              0.000000     2865.510              0.00   
2024-02-01      Santa Fe           2602.843196     2925.960           3059.54   
2024-02-01      Santa Fe              0.000000        0.000              0.00   

            OTRAS VENTAS   AXION ENERGY ARGENTINA S.A.  \
PERÍODO                                                  
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01         200.86                          0.0   
2010-01-01           3.53                          0.0   
...                   ...                          ...   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                       1080.5   
2024-02-01           0.00                          0.0   

            DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  PETROBRAS ARGENTINA S.A.  \
PERÍODO                                                                       
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
...                                           ...                       ...   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   

            ...  YPF S.A.  OIL COMBUSTIBLES S.A.  NEW AMERICAN OIL S.A.  \
PERÍODO     ...                                                           
2010-01-01  ...       0.0                    0.0

In [41]:
#Reemplazar valores NaN por 0:
df.fillna(0, inplace=True)
df

EMPRESA        LOCALIDAD  \
PERÍODO                                                                      
2010-01-01                ADVANCED ORGANIC MATERIALS S.A.            PILAR   
2010-01-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2010-01-01                                UNITEC BIO S.A.      SAN LORENZO   
2010-01-01                                     PITEY S.A.   VILLA MERCEDES   
2010-01-01                              VICENTÍN S.A.I.C.       AVELLANEDA   
...                                                   ...              ...   
2024-02-01  SANTA FE GAS Y ENERGÍAS RENOVABLES S.A.P.E.M.         SANTA FE   
2024-02-01                                 SOYENERGY S.A.            PILAR   
2024-02-01                            T 6 INDUSTRIAL S.A.      SAN LORENZO   
2024-02-01                                UNITEC BIO S.A.      SAN LORENZO   
2024-02-01                              VICENTÍN S.A.I.C.       AVELLANEDA   

               PROVINCIA  CUPO TOTAL ASIGNADO   PRODUCCIÓN   VENTAS AL CORTE   \
PERÍODO                                                                         
2010-01-01  BUENOS AIRES              0.000000      687.990              0.00   
2010-01-01      SANTA FE              0.000000    19175.270              0.00   
2010-01-01      SANTA FE              0.000000    15696.790              0.00   
2010-01-01      SAN LUIS              0.000000      201.017              0.00   
2010-01-01      SANTA FE              0.000000     8993.550              0.00   
...                  ...                   ...          ...               ...   
2024-02-01      Santa Fe              0.000000        0.000              0.00   
2024-02-01  Buenos Aires            937.173498      241.510            244.00   
2024-02-01      Santa Fe              0.000000     2865.510              0.00   
2024-02-01      Santa Fe           2602.843196     2925.960           3059.54   
2024-02-01      Santa Fe              0.000000        0.000              0.00   

            OTRAS VENTAS   AXION ENERGY ARGENTINA S.A.  \
PERÍODO                                                  
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01           0.00                          0.0   
2010-01-01         200.86                          0.0   
2010-01-01           3.53                          0.0   
...                   ...                          ...   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                          0.0   
2024-02-01           0.00                       1080.5   
2024-02-01           0.00                          0.0   

            DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  PETROBRAS ARGENTINA S.A.  \
PERÍODO                                                                       
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
2010-01-01                                    0.0                       0.0   
...                                           ...                       ...   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   
2024-02-01                                    0.0                       0.0   

            ...  YPF S.A.  OIL COMBUSTIBLES S.A.  NEW AMERICAN OIL S.A.  \
PERÍODO     ...                                                           
2010-01-01  ...       0.0                    0.0

In [42]:
#Top 5 últimos registros de compras en tn de biodiesel de las principales petroleras:
print(df[['AXION ENERGY ARGENTINA S.A.','PETROBRAS ARGENTINA S.A.','SHELL ARGENTINA C.A.P.S.A.','YPF S.A.','TRAFIGURA ARGENTINA S.A.']].head(-5))

            AXION ENERGY ARGENTINA S.A.  PETROBRAS ARGENTINA S.A.  \
PERÍODO                                                             
2010-01-01                         0.00                       0.0   
2010-01-01                         0.00                       0.0   
2010-01-01                         0.00                       0.0   
2010-01-01                         0.00                       0.0   
2010-01-01                         0.00                       0.0   
...                                 ...                       ...   
2024-02-01                       669.44                       0.0   
2024-02-01                         0.00                       0.0   
2024-02-01                         0.00                       0.0   
2024-02-01                      1406.60                       0.0   
2024-02-01                         0.00                       0.0   

            SHELL ARGENTINA C.A.P.S.A.  YPF S.A.  TRAFIGURA ARGENTINA S.A.  
PERÍODO                  

In [43]:
#Medidas de estadística descriptiva del df:
df.describe()

CUPO TOTAL ASIGNADO    PRODUCCIÓN   VENTAS AL CORTE   OTRAS VENTAS   \
count           5741.000000   5741.000000       5741.000000    5741.000000   
mean            2436.401690   5044.223326       2028.090675      30.054522   
std             1990.597431   8850.491235       2096.173487     324.198212   
min                0.000000      0.000000          0.000000       0.000000   
25%              550.000000    202.000000          0.000000       0.000000   
50%             2436.000000   2254.000000       1445.260000       0.000000   
75%             4166.000000   4291.900000       3891.760000       0.000000   
max            14500.000000  55405.600000      14512.000000    9682.610000   

       AXION ENERGY ARGENTINA S.A.  DESTILERÍA ARGENTINA DE PETRÓLEO S.A.  \
count                  5741.000000                            5741.000000   
mean                    304.111413                               1.681585   
std                     690.400331                              24.417294   
min                       0.000000                               0.000000   
25%                       0.000000                               0.000000   
50%                       0.000000                               0.000000   
75%                       0.000000                               0.000000   
max                    7438.160000                             857.220000   

       PETROBRAS ARGENTINA S.A.  PETROLERA DEL CONOSUR S.A.  POLIPETROL S.A.  \
count               5741.000000                 5741.000000      5741.000000   
mean                  73.959070                    8.543541         0.263567   
std                  292.542144                   82.539747         5.716578   
min                    0.000000                    0.000000         0.000000   
25%                    0.000000                    0.000000         0.000000   
50%                    0.000000                    0.000000         0.000000   
75%                    0.000000                    0.000000         0.000000   
max                 4829.030000                 1573.360000       179.000000   

       REFINERÍA DEL NORTE S.A.  ...      YPF S.A.  OIL COMBUSTIBLES S.A.  \
count               5741.000000  ...   5741.000000            5741.000000   
mean                  30.821756  ...   1186.015653              53.069394   
std                  158.708047  ...   1593.590469             197.927434   
min                    0.000000  ...      0.000000               0.000000   
25%                    0.000000  ...      0.000000               0.000000   
50%                    0.000000  ...    399.860000               0.000000   
75%                    0.000000  ...   1999.880000               0.000000   
max                 1781.340000  ...  14512.000000            2137.760000   

       NEW AMERICAN OIL S.A.  REFI PAMPA S.A.  TRAFIGURA ARGENTINA S.A.  \
count            5741.000000      5741.000000               5741.000000   
mean                2.225619         1.828901                 55.599737   
std                17.109422        15.988390                217.974556   
min                 0.000000         0.000000                  0.000000   
25%                 0.000000         0.000000                  0.000000   
50%                 0.000000         0.000000                  0.000000   
75%                 0.000000         0.000000                  0.000000   
max               313.650000       243.700000               2494.360000   

       ENERGÍA ARGENTINA S.A. (ENARSA)  ENERGÍA Y DERIVADOS DEL PETRÓLEO S.A.  \
count                      5741.000000                            5741.000000   
mean                          0.336596                               0.031120   
std                           8.659913                               0.962852   
min                           0.000000                               0.000000   
25%                           0.000000                               0.000000   
50%                           0.0

-Count: al reemplazar los valores NaN por 0, conseguimos que todas nuestras columnas tengan 5.741 registros o filas.
-Mean: la media nos marca medidas muy interesantes. Podemos observar: que los cupos asignados a las PYMES fueron en promedio 2436 tons, la producción por día promedio fue de 5.044 tons y las ventas en promedio superaron las 2.028 tons por día.
Otra información importante la analizamos por el lado de las compras de las grandes petroletas: YPF lidera las compras en promedio por día con 1.186 tons, seguida por Axion con 304 tons y Petrobras con 74 tons.
En la función "describe" podemos encontrar otras medidas de estadística descriptiva como el desvío estándar, mínimos, máximos, y análisis de cuartiles.